### PR TITLE
Log debug error if user cleanup is a not found error from missing `getent` binaries

### LINKED ISSUE #41918: Unactionable error message spam
Expected behavior:
This error doesn't appear if it's not something we're able to remediate.

Current behavior:
Seeing this error spammed in our auth / proxy cluster:
```
{
caller: "srv/usermgmt.go:469"
component: null
level: "error"
message: "Error during temporary user cleanup: cant find getent binary
	exec: "getent": executable file not found in $PATH"
timestamp: "2024-05-22T18:49:04Z"
}
```

Bug details:
- Teleport version: public.ecr.aws/gravitational/teleport-ent-distroless:15.1.9
- Recreation steps: Unknown, appears to have started once we cutover to the distroless images.

### PR DESCRIPTION
**Pull request by lxea**
Closes #41918

### CODE REVIEW

File: lib/srv/usermgmt.go [hunk 1]
```diff
@@ -466,7 +466,12 @@ func (u *HostUserManagement) UserCleanup() {
 	defer cleanupTicker.Stop()
 	for {
 		if err := u.DeleteAllUsers(); err != nil {
-			log.Error("Error during temporary user cleanup: ", err)
+			if trace.IsNotFound(err) {
```
Comment (zmb3) [inline lib/srv/usermgmt.go:469]: Suggestion:
```
switch err := u.DeleteAllUsers(); {
case trace.IsNotFound(err):
    log.Debugf("...")
    return
case err != nil:
    log.Error("...")

}
```

File: lib/srv/usermgmt.go
Comment (zmb3) [inline lib/srv/usermgmt.go:1]: Should our docs mention that host user operations won't work in distroless containers?

Should we check on startup whether the required tools are available and disable functionality that we know is not going to work?

Comment (lxea) [reply lib/srv/usermgmt.go:1]: Makes sense to me, pushed this

It doesn't mention it in the context of distroless containers but it does mention that it wont work unless the required binaries are in the path

**Review by gzdunek — APPROVED** [review]

**Comment by lxea** [comment]
@zmb3 do you mind giving this another quick look over, thanks!

**Review by zmb3 — APPROVED** [review]

**Pull request merged**